### PR TITLE
9757 Orchestration docs timebox

### PR DESCRIPTION
**Pull request by turadg**
refs: #9757

## Description

A bunch of docs improvements to get into the u17 release.

This doesn't do all the things specified in the ticket so leaves open:
- #9757 

### Security Considerations
n/a

### Scaling Considerations
n/a

### Documentation Considerations
We'll need tutorial docs. I think some of the `@example` suggestions make more sense there.

### Testing Considerations
CI

### Upgrade Considerations
not yet deployed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "prettier-plugin-jsdoc": "^1.3.0",
     "prettier-plugin-sh": "^0.14.0",
     "type-coverage": "^2.27.1",
-    "typedoc": "^0.26.4",
+    "typedoc": "^0.26.7",
```
Comment (dckc) [inline package.json:35]: are there any particular features that we're using from this release? or was this just housekeeping?

I notice it's less than a day old. For ref: c56c478

File: packages/orchestration/src/utils/start-helper.js [hunk 2]
```diff
@@ -176,6 +176,13 @@ harden(provideOrchestration);
  * Simplifies contract functions for Orchestration by wrapping a simpler
  * function with all the tools it needs in order to use Orchestration.
  *
+ * @example
+ *
+ * ```js
+ * const contract = (zcf, privateArgs, zone, tools) => { ... };
+ * export const start = withOrchestration(contract);
+ * ```
+ *
```
Comment (dckc) [inline packages/orchestration/src/utils/start-helper.js:185]: I don't see `withOrchestration` on https://9757-timebox.agoric-sdk.pages.dev/modules/_agoric_orchestration :-/

File: packages/orchestration/src/cosmos-api.ts [hunk 3]
```diff
@@ -141,6 +147,13 @@ export interface StakingAccountQueries {
    */
   getReward: (validator: CosmosValidatorAddress) => Promise<DenomAmount[]>;
 }
+
+/**
+ * Transactions for doing staking operations on an individual account.
+ *
+ * @see {@link https://docs.cosmos.network/main/build/modules/staking#messages x/staking messages}
```
Comment (dckc) [inline packages/orchestration/src/cosmos-api.ts]: a comma or something between the links would help.

it's a little hard to tell that there are 2 of them.

File: packages/orchestration/src/cosmos-api.ts [hunk 4]
```diff
@@ -263,12 +276,26 @@ export type LocalAccountMethods = {
...
   timeoutHeight?: MsgTransfer['timeoutHeight'];
   timeoutTimestamp?: MsgTransfer['timeoutTimestamp'];
   memo?: string;
 };
 
+/**
+ * Cosmos-specific methods to extend `OrchestrationAccountI`, parameterized
+ * by `CosmosChainInfo`.
+ *
+ * In particular, if the chain info includes a staking token, `StakingAccountActions`
```
Comment (dckc) [inline packages/orchestration/src/cosmos-api.ts]: use `{@link StakingAccountActions}`?

File: packages/orchestration/src/orchestration-api.ts [hunk 5]
```diff
@@ -234,6 +234,9 @@ export interface OrchestrationAccountI {
   getPublicTopics: () => Promise<Record<string, ResolvedPublicTopic<unknown>>>;
 }
 
+/**
+ * An async-flow suitable for use in contracts using Orchestration
```
Comment (dckc) [inline packages/orchestration/src/orchestration-api.ts]: "async-flow" is a pretty opaque term here.

I suggest moving (most of) the "Orchestration flows" section of the README to here.

File: packages/orchestration/src/orchestration-api.ts [hunk 6]
```diff
@@ -246,6 +246,7 @@ export interface OrchestrationFlow<CT = unknown> {
  * The type must be able to express transfers across different chains and transports.
  *
  * NOTE Expected to change, so consider an opaque structure.
+ * @internal
```
Comment (dckc) [inline packages/orchestration/src/orchestration-api.ts:266]: looks like you got this working. nice!

File: packages/orchestration/src/exos/chain-hub.js [hunk 7]
```diff
@@ -40,11 +41,24 @@ export const DenomDetailShape = M.splitRecord(
   { brand: BrandShape },
 );
 
-/** agoricNames key for ChainInfo hub */
+// TODO refactor into an enum-ish object
+/**
+ * agoricNames key for ChainInfo hub
+ *
+ * @internal
```
Comment (dckc) [inline packages/orchestration/src/exos/chain-hub.js]: These agoricNames keys don't seem internal to me. They show up all the way to clients via vstorage.

File: packages/orchestration/src/exos/ica-account-kit.js [hunk 8]
```diff
@@ -42,7 +41,9 @@ export const IcaAccountI = M.interface('IcaAccount', {
   reactivate: M.call().returns(VowShape),
 });
 
+// XXX none of these modifiers are working to exclude this type from api-docs
```
Comment (dckc) [inline packages/orchestration/src/exos/ica-account-kit.js:44]: oh. so it's _not_ yet working?

File: packages/orchestration/src/cosmos-api.ts [hunk 9]
```diff
@@ -36,7 +36,7 @@ export type CosmosValidatorAddress = ChainAddress & {
 };
 
 /** Represents an IBC Connection between two chains, which can contain multiple Channels. */
-export type IBCConnectionInfo = {
+export interface IBCConnectionInfo {
```
Comment (dckc) [inline packages/orchestration/src/cosmos-api.ts:39]: I'd expect to use interfaces for things with methods, but this is an ordinary data structure. Why not leave it as a type alias?

File: packages/orchestration/src/cosmos-api.ts [hunk 10]
```diff
@@ -56,12 +56,12 @@ export type IBCConnectionInfo = {
...
     version: string; // e.eg. 'ics20-1'
   };
-};
+}
 
 /**
  * https://github.com/cosmos/chain-registry/blob/master/assetlist.schema.json
  */
-export type CosmosAssetInfo = {
+export interface CosmosAssetInfo extends Record<string, unknown> {
```
Comment (dckc) [inline packages/orchestration/src/cosmos-api.ts:64]: likewise, seems fine as a type alias

File: packages/orchestration/src/cosmos-api.ts (hunk 9 above)
Comment (dckc) [reply packages/orchestration/src/cosmos-api.ts:39]: I guess being able to refer to each property with a link is nice.

File: package.json (hunk 1 above)
Comment (dckc) [reply package.json:35]: maybe hoping for better visibility support?

**Review by dckc — APPROVED** [review]
Pretty clearly forward progress.

The closest thing to a critical comment is on marking the agoricNames keys as internal.But it's conservative - we can expose it later.

File: packages/orchestration/src/utils/start-helper.js (hunk 2 above)
Comment (turadg) [reply packages/orchestration/src/utils/start-helper.js:185]: oh yeah, that got stalled: https://github.com/Agoric/agoric-sdk/pull/9808/

I'll try to solve that here

File: packages/orchestration/src/cosmos-api.ts (hunk 3 above)
Comment (turadg) [reply packages/orchestration/src/cosmos-api.ts:154]: oh I `}` has to go after the URL ends. I'll fix that

File: packages/orchestration/src/exos/chain-hub.js (hunk 7 above)
Comment (turadg) [reply packages/orchestration/src/exos/chain-hub.js]: good point, I'll revert those. I still think they should go into an enum, but that can wait

File: packages/orchestration/src/exos/ica-account-kit.js (hunk 8 above)
Comment (turadg) [reply packages/orchestration/src/exos/ica-account-kit.js:44]: in some cases, no. maybe on types that are transitively included

File: packages/orchestration/src/cosmos-api.ts (hunk 9 above)
Comment (turadg) [reply packages/orchestration/src/cosmos-api.ts:39]: I read some perf docs recently: https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections

File: packages/orchestration/src/exos/ica-account-kit.js (hunk 8 above)
Comment (turadg) [reply packages/orchestration/src/exos/ica-account-kit.js:44]: in some cases, no. maybe on types that are transitively included

File: packages/orchestration/src/chain-info.js [hunk 11]
```diff
@@ -9,10 +9,14 @@ import {
 import fetchedChainInfo from './fetched-chain-info.js'; // Refresh with scripts/refresh-chain-info.ts
 import { CosmosAssetInfoShape, CosmosChainInfoShape } from './typeGuards.js';
 
-/** @import {CosmosAssetInfo, CosmosChainInfo, EthChainInfo, IBCConnectionInfo} from './types.js'; */
+/** @import {Chain, CosmosAssetInfo, CosmosChainInfo, EthChainInfo, IBCConnectionInfo} from './types.js'; */
 /** @import {NameAdmin} from '@agoric/vats'; */
 
-/** @typedef {CosmosChainInfo | EthChainInfo} ChainInfo */
+/**
+ * Info used to build a {@link Chain} object - naming, connections, etc.
```
Comment (0xpatrickdev) [inline packages/orchestration/src/chain-info.js]: nit: `naming` stuck out to me as something to improve. maybe:
```suggestion
 * Info used to build a {@link Chain} object - channel, connection, and denom info.
```

File: packages/orchestration/src/orchestration-api.ts (hunk 5 above)
Comment (turadg) [reply packages/orchestration/src/orchestration-api.ts]: good idea!

File: packages/orchestration/src/chain-info.js [hunk 12]
```diff
@@ -78,12 +78,14 @@ const knownChains = /** @satisfies {Record<string, ChainInfo>} */ (
  * @internal
  */
 
+// TODO(#9572): include this in registerChain
```
Comment (0xpatrickdev) [inline packages/orchestration/src/chain-info.js:82]: 👏

File: packages/orchestration/src/typeGuards.js [hunk 13]
```diff
@@ -133,6 +134,7 @@ export const DenomAmountShape = { denom: DenomShape, value: M.bigint() };
 /** @type {TypedPattern<AmountArg>} */
 export const AmountArgShape = M.or(AmountShape, DenomAmountShape);
 
+/** @type {TypedPattern<RequestQuery>} */
```
Comment (0xpatrickdev) [inline packages/orchestration/src/typeGuards.js]: `RequestQuery['data']` is `Uint8Array`. Was going to suggest using `TypedPattern<JsonSafe<RequestQuery>>`, but we need to preserve `RequestQuery['height']: bigint`

**Review by 0xpatrickdev — APPROVED** [review]

File: packages/orchestration/src/typeGuards.js (hunk 13 above)
Comment (turadg) [reply packages/orchestration/src/typeGuards.js]: thanks, I'll correct this

**Pull request merged**